### PR TITLE
feat: Implement geometry loader

### PR DESCRIPTION
**Pull request by henriknorin-ftrack**
Resolves : 

* CLICKUP-3vta7gk

- [ ] I have added automatic tests where applicable.
- [ ] The PR contains a description of what has been changed.
- [ ] The description contains manual test instructions.
- [ ] The PR contains update to the release notes.
- [ ] The PR contains update to the documentation.

This PR has been tested on :

- [X] Windows.
- [ ] MacOs.
- [ ] Linux.

## Changes

Implemented and tested FBX geometry loader.
Prepared rest of loaders.

## Test

_NOTE: Has to be tested on corresponding branch on definition and unreal._

Publish FBX geometry in Maya and test loading it and asset manage it in Unreal, it should work as expected.

### CODE REVIEW

File: resource/plugins/unreal/python/loader/importers/unreal_abc_animation_loader_importer.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+        task.filename = component_path
+
+        asset_version_entity = self.session.query(
+            'AssetVersion where id={}'.format(context_data['version_id'])
+        ).first()
+        import_path = (
+            '/Game/'
+            + unreal_utils.get_context_relative_path(
+                self.session, asset_version_entity['task']
```
Comment (lluisCM) [inline resource/plugins/unreal/python/loader/importers/unreal_abc_animation_loader_importer.py:43]: Reformat this using our standards. ex: "/Game/{}/{}".format(....."

File: resource/plugins/unreal/python/loader/importers/unreal_abc_animation_loader_importer.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+            )
+            + context_data['asset_name']
+        )
+
+        task.destination_path = import_path.replace(' ', '_')
```
Comment (lluisCM) [inline resource/plugins/unreal/python/loader/importers/unreal_abc_animation_loader_importer.py:48]: IS this safe? we are replacing spaces to underscore in the path, so the path will not be valid anymore... right?

File: resource/plugins/unreal/python/loader/importers/unreal_abc_animation_loader_importer.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+        task.replace_existing = options.get('ReplaceExisting', True)
+        task.automated = options.get('Automated', True)
+        task.save = options.get('Save', True)
+
+        import_result = unreal_utils.import_file(task)
+        self.logger.info(
+            'Imported Alembic animation: {}'.format(import_result)
+        )
+        loaded_anim = unreal.EditorAssetLibrary.load_asset(import_result)
```
Comment (lluisCM) [inline resource/plugins/unreal/python/loader/importers/unreal_abc_animation_loader_importer.py:92]: This is not needed

File: resource/plugins/unreal/python/loader/importers/unreal_abc_animation_loader_importer.py [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+        results = {}
+
+        results[component_path] = unreal_utils.rename_node_with_prefix(
```
Comment (lluisCM) [inline resource/plugins/unreal/python/loader/importers/unreal_abc_animation_loader_importer.py:96]: Why we re renaming?

File: resource/plugins/unreal/python/loader/importers/unreal_abc_geometry_loader_importer.py [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+            paths_to_import.extend(collector['result'])
+
+        component_path = paths_to_import[0]
+
+        task.filename = component_path
+
+        asset_version_entity = self.session.query(
+            'AssetVersion where id={}'.format(context_data['version_id'])
+        ).first()
+        import_path = (
```
Comment (lluisCM) [inline resource/plugins/unreal/python/loader/importers/unreal_abc_geometry_loader_importer.py:38]: Use our standards

File: resource/plugins/unreal/python/loader/importers/unreal_abc_geometry_loader_importer.py [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+            '/Game/'
+            + unreal_utils.get_context_relative_path(
+                self.session, asset_version_entity['task']
+            )
+            + context_data['asset_name']
+        )
+
+        task.destination_path = import_path.replace(' ', '_')
```
Comment (lluisCM) [inline resource/plugins/unreal/python/loader/importers/unreal_abc_geometry_loader_importer.py:46]: Not sure why are we replacing space?

File: resource/plugins/unreal/python/loader/importers/unreal_abc_geometry_loader_importer.py [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+        task.save = options.get('Save', True)
+
+        import_result = unreal_utils.import_file(task)
+
+        if import_result is None:
+            raise Exception('Alembic geo import failed!')
+
+        self.logger.info('Imported Alembic geo: {}'.format(import_result))
+
+        loaded_mesh = unreal.EditorAssetLibrary.load_asset(import_result)
```
Comment (lluisCM) [inline resource/plugins/unreal/python/loader/importers/unreal_abc_geometry_loader_importer.py:71]: Why loading?

File: resource/plugins/unreal/python/loader/importers/unreal_abc_geometry_loader_importer.py [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+
+        results = {}
+
+        results[component_path] = unreal_utils.rename_node_with_prefix(
```
Comment (lluisCM) [inline resource/plugins/unreal/python/loader/importers/unreal_abc_geometry_loader_importer.py:75]: Why renaming?

File: resource/plugins/unreal/python/loader/importers/unreal_abc_rig_loader_importer.py [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+        asset_version_entity = self.session.query(
+            'AssetVersion where id={}'.format(context_data['version_id'])
+        ).first()
+        import_path = (
+            '/Game/'
+            + unreal_utils.get_context_relative_path(
+                self.session, asset_version_entity['task']
+            )
+            + context_data['asset_name']
+        )
```
Comment (lluisCM) [inline resource/plugins/unreal/python/loader/importers/unreal_abc_rig_loader_importer.py:44]: Use our standards

File: resource/plugins/unreal/python/loader/importers/unreal_abc_rig_loader_importer.py [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+
+        task.destination_path = import_path.replace(' ', '_')
```
Comment (lluisCM) [inline resource/plugins/unreal/python/loader/importers/unreal_abc_rig_loader_importer.py:46]: Why replacing?

File: resource/plugins/unreal/python/loader/importers/unreal_abc_rig_loader_importer.py [hunk 11]
```diff
@@ -0,0 +1,113 @@
...
+                    'skeleton', skeletonAD.get_asset()
+                )
+
+        task.replace_existing = options.get('ReplaceExisting', True)
+        task.automated = options.get('Automated', True)
+        task.save = options.get('Save', True)
+
+        import_result = unreal_utils.import_file(task)
+        self.logger.info('Imported Alembic rig: {}'.format(import_result))
+        loaded_skeletal_mesh = unreal.EditorAssetLibrary.load_asset(
```
Comment (lluisCM) [inline resource/plugins/unreal/python/loader/importers/unreal_abc_rig_loader_importer.py:82]: Why loading an asset?

File: resource/plugins/unreal/python/loader/importers/unreal_abc_rig_loader_importer.py [hunk 12]
```diff
@@ -0,0 +1,113 @@
...
+        mesh_skeleton = loaded_skeletal_mesh.skeleton
+        if mesh_skeleton:
+            results[component_path].append(
+                self._rename_object_with_prefix(mesh_skeleton, 'SKEL')
+            )
+
+        mesh_physics_asset = loaded_skeletal_mesh.physics_asset
+        if mesh_physics_asset:
+            results[component_path].append(
+                self._rename_object_with_prefix(mesh_physics_asset, 'PHAT')
```
Comment (lluisCM) [inline resource/plugins/unreal/python/loader/importers/unreal_abc_rig_loader_importer.py:101]: why rename with prefix?

File: resource/plugins/unreal/python/loader/importers/unreal_fbx_animation_loader_importer.py [hunk 13]
```diff
@@ -0,0 +1,132 @@
...
+        asset_version_entity = self.session.query(
+            'AssetVersion where id={}'.format(context_data['version_id'])
+        ).first()
+        import_path = (
+            '/Game/'
+            + unreal_utils.get_context_relative_path(
+                self.session, asset_version_entity['task']
+            )
+            + context_data['asset_name']
+        )
```
Comment (lluisCM) [inline resource/plugins/unreal/python/loader/importers/unreal_fbx_animation_loader_importer.py:46]: Repplace with our standards

File: resource/plugins/unreal/python/loader/importers/unreal_fbx_animation_loader_importer.py [hunk 14]
```diff
@@ -0,0 +1,132 @@
...
+
+        task.destination_path = import_path.replace(' ', '_')
```
Comment (lluisCM) [inline resource/plugins/unreal/python/loader/importers/unreal_fbx_animation_loader_importer.py:48]: Why replace?

File: resource/plugins/unreal/python/loader/importers/unreal_fbx_animation_loader_importer.py [hunk 15]
```diff
@@ -0,0 +1,132 @@
...
+                    'skeleton', skeletonAD.get_asset()
+                )
+
+        task.replace_existing = options.get('ReplaceExisting', True)
+        task.automated = options.get('Automated', True)
+        task.save = options.get('Save', True)
+
+        import_result = unreal_utils.import_file(task)
+        self.logger.info('Imported FBX animation: {}'.format(import_result))
+        loaded_anim = unreal.EditorAssetLibrary.load_asset(import_result)
```
Comment (lluisCM) [inline resource/plugins/unreal/python/loader/importers/unreal_fbx_animation_loader_importer.py:115]: Why loading?

File: resource/plugins/unreal/python/loader/importers/unreal_fbx_animation_loader_importer.py [hunk 16]
```diff
@@ -0,0 +1,132 @@
...
+
+        results = {}
+
+        results[component_path] = unreal_utils.rename_node_with_prefix(
+            loaded_anim, 'A'
+        )
```
Comment (lluisCM) [inline resource/plugins/unreal/python/loader/importers/unreal_fbx_animation_loader_importer.py:121]: why rename?

File: resource/plugins/unreal/python/loader/importers/unreal_fbx_geometry_loader_importer.py [hunk 17]
```diff
@@ -0,0 +1,95 @@
...
+
+        asset_version_entity = self.session.query(
+            'AssetVersion where id={}'.format(context_data['version_id'])
+        ).first()
+        import_path = (
+            '/Game/'
+            + unreal_utils.get_context_relative_path(
+                self.session, asset_version_entity['task']
+            )
+            + context_data['asset_name']
```
Comment (lluisCM) [inline resource/plugins/unreal/python/loader/importers/unreal_fbx_geometry_loader_importer.py:43]: Use our standards

File: resource/plugins/unreal/python/loader/importers/unreal_fbx_geometry_loader_importer.py [hunk 18]
```diff
@@ -0,0 +1,95 @@
...
+        )
+
+        task.destination_path = import_path.replace(' ', '_')
```
Comment (lluisCM) [inline resource/plugins/unreal/python/loader/importers/unreal_fbx_geometry_loader_importer.py:46]: Why replace?

File: resource/plugins/unreal/python/loader/importers/unreal_fbx_geometry_loader_importer.py [hunk 19]
```diff
@@ -0,0 +1,95 @@
...
+
+        # Geometry specific options
+
+        task.replace_existing = options.get('ReplaceExisting', True)
+        task.automated = options.get('Automated', True)
+        task.save = options.get('Save', True)
+
+        import_result = unreal_utils.import_file(task)
+        self.logger.info('Imported FBX geo: {}'.format(import_result))
+        loaded_mesh = unreal.EditorAssetLibrary.load_asset(import_result)
```
Comment (lluisCM) [inline resource/plugins/unreal/python/loader/importers/unreal_fbx_geometry_loader_importer.py:78]: No need to load

File: resource/plugins/unreal/python/loader/importers/unreal_fbx_geometry_loader_importer.py [hunk 20]
```diff
@@ -0,0 +1,95 @@
...
+
+        results = {}
+
+        results[component_path] = unreal_utils.rename_node_with_prefix(
+            loaded_mesh, 'S'
+        )
```
Comment (lluisCM) [inline resource/plugins/unreal/python/loader/importers/unreal_fbx_geometry_loader_importer.py:84]: why rename?

File: resource/plugins/unreal/python/loader/importers/unreal_fbx_rig_loader_importer.py [hunk 21]
```diff
@@ -0,0 +1,119 @@
...
+        asset_version_entity = self.session.query(
+            'AssetVersion where id={}'.format(context_data['version_id'])
+        ).first()
+        import_path = (
+            '/Game/'
+            + unreal_utils.get_context_relative_path(
+                self.session, asset_version_entity['task']
+            )
+            + context_data['asset_name']
+        )
```
Comment (lluisCM) [inline resource/plugins/unreal/python/loader/importers/unreal_fbx_rig_loader_importer.py:46]: Standards

File: resource/plugins/unreal/python/loader/importers/unreal_fbx_rig_loader_importer.py [hunk 22]
```diff
@@ -0,0 +1,119 @@
...
+
+        task.destination_path = import_path.replace(' ', '_')
```
Comment (lluisCM) [inline resource/plugins/unreal/python/loader/importers/unreal_fbx_rig_loader_importer.py:48]: ??

File: resource/plugins/unreal/python/loader/importers/unreal_fbx_rig_loader_importer.py [hunk 23]
```diff
@@ -0,0 +1,119 @@
...
+                    'skeleton', skeletonAD.get_asset()
+                )
+
+        task.replace_existing = options.get('ReplaceExisting', True)
+        task.automated = options.get('Automated', True)
+        task.save = options.get('Save', True)
+
+        import_result = unreal_utils.import_file(task)
+        self.logger.info('Imported FBX rig: {}'.format(import_result))
+        loaded_mesh = unreal.EditorAssetLibrary.load_asset(import_result)
```
Comment (lluisCM) [inline resource/plugins/unreal/python/loader/importers/unreal_fbx_rig_loader_importer.py:102]: No need to load

File: resource/plugins/unreal/python/loader/importers/unreal_fbx_rig_loader_importer.py [hunk 24]
```diff
@@ -0,0 +1,119 @@
...
+
+        results = {}
+
+        results[component_path] = unreal_utils.rename_node_with_prefix(
+            loaded_mesh, 'S'
+        )
```
Comment (lluisCM) [inline resource/plugins/unreal/python/loader/importers/unreal_fbx_rig_loader_importer.py:108]: why?

**Review by lluisCM — CHANGES_REQUESTED** [review]
Please fix the all the importers (I just commented some of them) I'll review them all once fixed. Thanks

File: resource/plugins/unreal/python/loader/importers/unreal_abc_animation_loader_importer.py (hunk 1 above)
Comment (henriknorin-ftrack) [reply resource/plugins/unreal/python/loader/importers/unreal_abc_animation_loader_importer.py:43]: Done in next PR (finalise loaders)

File: resource/plugins/unreal/python/loader/importers/unreal_abc_animation_loader_importer.py (hunk 2 above)
Comment (henriknorin-ftrack) [reply resource/plugins/unreal/python/loader/importers/unreal_abc_animation_loader_importer.py:48]: This is the destination path, Unreal does not support space in folders/file names.

File: resource/plugins/unreal/python/loader/importers/unreal_abc_animation_loader_importer.py (hunk 4 above)
Comment (henriknorin-ftrack) [reply resource/plugins/unreal/python/loader/importers/unreal_abc_animation_loader_importer.py:96]: Will put as option

File: resource/plugins/unreal/python/loader/importers/unreal_abc_animation_loader_importer.py (hunk 3 above)
Comment (henriknorin-ftrack) [reply resource/plugins/unreal/python/loader/importers/unreal_abc_animation_loader_importer.py:92]: Will try remove this and see if it works as expected.

File: resource/plugins/unreal/python/loader/importers/unreal_abc_geometry_loader_importer.py (hunk 5 above)
Comment (henriknorin-ftrack) [reply resource/plugins/unreal/python/loader/importers/unreal_abc_geometry_loader_importer.py:38]: Will be resolved in next PR

**Pull request merged**